src/extra/components/timepix.py

```diff
@@ -441,6 +441,21 @@ def pixel_events(self, pulse_dim='pulseId', toa_offset=0.0,
         hits_per_train = self.raw_size_key.ndarray()
         num_hits = int(hits_per_train.sum())
 
+        if (hits_per_train > self.raw_x_key.entry_shape[0]).any():
+            # data.size may actually record more pixel events than can
+            # be placed into array buffer. The device is supposed to
+            # prevent this, but there is some commissioning data
```

Comment:
Does the fixed device still record somewhere that some hits were dropped because of the buffer size? It seems worth having some way to surface that.

Reply:
Yes, as I understand from @bj-s the device is supposed to go into error in this situation.